### PR TITLE
Update guideline Correct Format for Arrays

### LINKED ISSUE #444: StyleGuidelines.md: Line Character Limit?
Within the 'StyleGuidelines.md' file in the 'Correct Format for Arrays' section, it states: "Arrays should be written on one line unless they exceed the line character limit". However, I can't find anywhere in the style guidelines that actually say what the line character limit is. Can this be clarified in the documentation? I'm not sure if this is referring to a hard character limit for files, or if this is a style limit that should be followed. Thanks.

### PR DESCRIPTION
**Pull request by johlju**
#### Pull Request (PR) description
- Update guideline Correct Format for Arrays  …
  - Updated text around line length (issue #444).
  - Added alternative good ways of declaring an array with multiple items.

#### This Pull Request (PR) fixes the following issues
- Fixes #444 

#### Task list

- [x] This PR changes documentation
- [ ] This PR changes DscResource.Template
- [ ] This PR changes Tests.Template
  - [ ] The template version number has been increased appropriately.
  - [ ] An descriptive entry has been added to the CHANGELOG.md under the
        Tests.Template folder.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/dscresources/445)

### CODE REVIEW

**Comment by johlju** [comment]
@mhendric @PlagueHO could you both review this proposal to see if this is a change we should do (and proof read please 🙂)

**Comment by mhendric** [comment]
Hi @johlju , in addition to the examples, it may be good to describe what a Good multi-line array declaration looks like. I'm also wondering if the hashtable example should be separate from (or in addition to) the multi-line array example. Finally, it may be good to add a couple Bad examples showing what a bad multi-line array looks like, and what declaring a hashtable within an array looks like.

**Comment by johlju** [comment]
>  it may be good to describe what a Good multi-line array declaration looks like.  I'm also wondering if the hashtable example should be separate from (or in addition to) the multi-line array example

@mhendric what do you mean by a 'multi-line array'? 

> Finally, it may be good to add a couple Bad examples showing what a bad multi-line array looks like, and what declaring a hashtable within an array looks like.

Updated the examples a bit.

File: StyleGuidelines.md [hunk 1]
```diff
@@ -167,20 +167,92 @@ $superLongVariableName = Get-MySuperLongVariablePlease @getMySuperLongVariablePl
...
+$myArray = @(
+    @{Key1 = Value1;Key2 = Value2},
+    @{Key1 = Value1;Key2 = Value2}
+)
+```
+
+**Good:**
+
+```powershell
+$array = @( 'one', 'two', 'three')
```
Comment (johlju) [inline StyleGuidelines.md]: Found a bug here, spaces are not consistent

Comment (johlju) [reply StyleGuidelines.md]: Done.

**Comment by mhendric** [comment]
>@mhendric what do you mean by a 'multi-line array'?

@johlju , sorry for the late response. By this I just meant an array declaration where each element is defined on a separate line.

**Comment by johlju** [comment]
@mhendric I think I got an example of that? But if you think an example is missing then please submit an PR for that :)

**Pull request merged**